timemachine/fe/rbfe.py

```diff
@@ -135,18 +135,18 @@ def sample(initial_state, protocol):
 
     # a crude, and probably not great, guess on the decorrelation time
     n_steps = protocol.n_frames * protocol.steps_per_frame
-    all_nrgs, all_coords, all_boxes = ctxt.multiple_steps_U(
+    _, all_coords, all_boxes = ctxt.multiple_steps_U(
         lamb=initial_state.lamb,
         n_steps=n_steps,
-        lambda_windows=[initial_state.lamb],
-        store_u_interval=protocol.steps_per_frame,
+        lambda_windows=[],
+        store_u_interval=0,
         store_x_interval=protocol.steps_per_frame,
     )
 
     assert all_coords.shape[0] == protocol.n_frames
     assert all_boxes.shape[0] == protocol.n_frames
 
-    assert np.all(np.isfinite(ctxt.get_x_t())), "Production resulted in a nan"
+    assert np.all(np.isfinite(all_coords[-1])), "Production resulted in a nan"
```

Comment:
This avoids allocating a new array, just use the coordinates that were returned.